### DISCUSSION TITLE
Contrast adjustment to a fyne canvas.Image

### DISCUSSION THREAD

**Discussion by bob-anderson-ok** [fyne-io/fyne#4657] [Q&A] [answered]
How can I add contrast adjustment to a canvas.Image

I have four types to deal with:  

image.Gray  image.Gray16 (I have solved these types already)

fltimg.Gray32 and fltimg.Gray64 have me stumped.

**Comment by andydotxyz** [comment, marked as answer]
We don't have built-in colour controls, so it's a case of modifying pixel values with some image manipulation library.
I'm not familiar with `fltimg`, perhaps they can help with colourising the pixels.

**Comment by bob-anderson-ok** [comment]
Thanks. I have done pixel value manipulation on the 8 bit and 16 bit images. I'll probably end up doing the smae on the float32 and float64 images as well.
